### DISCUSSION TITLE
[Schema] Can you set a structured conf attribute as optional?

### DISCUSSION THREAD

**Discussion by JuanPabloMF** [facebookresearch/hydra#2687] [Q&A]
I'm using dataclasses as schema for my config.

```
@dataclass
class ModelConfig:
    path: str = "models"
    tresh: float = 0.5
    max_detections: Optional[int] = None
```

This schema lets me define a resnet.yaml like so:

```
defaults:
  - /schemas/base_model_config@_here_

- path: "new_models"
- tresh: 0.6 
```

Giving as final result:

```
- path: "new_models"
- tresh: 0.6
- max_detections: null
```

Is there any way to still check that max_detections is an integer (if defined), but if not defined to not add it to my final yaml. Currently I'm having a lot of default fields filled with None (the real config is a much larger and composed yaml).

I was thinking of __post_init__ together with InitVar, but as you can check in https://github.com/facebookresearch/hydra/issues/981 this doesn't seem to be possible.

**Comment by omry** [comment]
Not really.
